main.js

```diff
@@ -178,4 +178,383 @@ const data = [
+            if (!deepEqual(object1_value, object2_value))
+                return false;
+        } else if (object1_value !== object2_value) return false;
+        object1_value = object1[object2_names[i]];
+        object2_value = object2[object2_names[i]];
+        if ((typeof object1_value) != (typeof object2_value))
+            return false;
+        if (object1_value instanceof Object) {
+            if (!deepEqual(object1_value, object2_value))
+                return false;
+        } else if (object1_value !== object2_value)
+            return false;
+    }
+    return true;
+}
+console.log(deepEqual(obj1,obj2));
+console.log(deepEqual(obj1,obj3));
+
+function getCalendarMonth(daysInMonth, daysInWeek, dayOfWeek, checkInDate, checkOutDate) {
+    if (dayOfWeek >= daysInWeek)
+        throw Error(`${dayOfWeek} >= ${daysInWeek}`)
+    if (checkInDate > daysInMonth || checkOutDate > daysInMonth)
+        throw Error('Check in range is invalid')
+    if (checkOutDate < checkInDate)
+        throw Error(`${checkOutDate} < ${checkInDate}`)
+
+    let month = []
+    let x = (daysInWeek - ((daysInMonth + dayOfWeek) % daysInWeek)) % daysInWeek // count of days to complete last week.
+    for (let i = 0; i < daysInMonth + dayOfWeek + x; i++) {
+        if (i % daysInWeek == 0) {
+            month.push([])
+        }
+        let last = month.length - 1
+        if (i < dayOfWeek) {
+            //get last element in array
+            month[last].push({
+                dayOfMonth: daysInMonth - dayOfWeek + i + 1,
+                notCurrentMonth: true,
+                selectedDay: false,
+
+            })
+        } else if (i >= daysInMonth + dayOfWeek) {
+            month[last].push({
+                dayOfMonth: i % daysInMonth - dayOfWeek + 1,
+                notCurrentMonth: true,
+                selectedDay: false,
+            })
+        } else {
+            let date = i - dayOfWeek + 1;
+            month[last].push({
+                dayOfMonth: date,
+                notCurrentMonth: false,
+                selectedDay: (date >= checkInDate && date <= checkOutDate),
+            });
+        }
+    }
+    return month;
+}
+console.log(getCalendarMonth(30, 7, 4, 17, 21));
+
+// lesson-8
+class User {
+    constructor(firstName, lastName) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+
+    }
+    get fullName () {
+        return `${this.firstName} ${this.lastName}`;
+    }
+
+}
+
+class Student extends User {
+    constructor(firstName, lastName, admissionYear, courseName) {
+        super(firstName, lastName);
+        this.admissionYear = admissionYear;
+        this.courseName = courseName;
+    }
+    get course () {
+
+        let currentYear = new Date().getFullYear();
+        return currentYear - this.admissionYear;
+    }
+}
+
+class Students {
+    constructor(students){
+        this.students = students;
+    }
+}
+Students.prototype.getInfo = function (){
```

Comment:
it's not necessary to write Students.prototype here, you can just write getInfo() {...}

Reply:
this will be a method in a class with will act differently for each instance depending on 'this'

Reply:
everything else in this hometask is great! Amazing job!

Reply:
Using Students.getInfo = function() { … } we’re defining a static function, but task says getInfo() should be instance’s function. getInfo() could be defined in a class, but according task text getInfo should be defined as «prototype, function». Maybe I’ve missunderstood the task